File: QubitServer/src/main/java/org/labrad/qubits/mem/NoopCommand.java
package org.labrad.qubits.mem;

import org.labrad.qubits.FpgaModelDac;

public class NoopCommand implements MemoryCommand {
  private NoopCommand() {}

  private static final NoopCommand INSTANCE = new NoopCommand();

  public static NoopCommand getInstance() {
    return INSTANCE;
  }

  public long[] getBits() {
    return new long[] {0x000000};
  }
  public double getTime_us(FpgaModelDac dac) {
    return FpgaModelDac.clocksToMicroseconds(1);
  }
}

File: QubitServer/src/main/scala/org/labrad/qubits/mem/NoopCommand.scala
package org.labrad.qubits.mem

import org.labrad.qubits.FpgaModelDac

object NoopCommand extends MemoryCommand {
  def cmdBits: Array[Long] = {
    Array[Long](0x000000)
  }

  def time_us(dac: FpgaModelDac): Double = {
    FpgaModelDac.clocksToMicroseconds(1)
  }
}
